server/platform/services/sharedchannel/channelinvite.go

```diff
@@ -41,16 +42,24 @@ func (cim channelInviteMsg) DirectParticipantsMap() map[string]*model.User {
 
 type InviteOption func(msg *channelInviteMsg)
 
-func WithDirectParticipant(participant *model.User) InviteOption {
+func WithDirectParticipant(participant *model.User, remoteID string) InviteOption {
 	return func(msg *channelInviteMsg) {
 		msg.DirectParticipantIDs = append(msg.DirectParticipantIDs, participant.Id)
-		// if the participant is local, send it as part of the invite payload
-		if !participant.IsRemote() {
+		// if the participant doesn't belong to the remote we're
+		// sending the invite to, send it as part of the invite
+		// payload
+		if participant.GetRemoteID() != remoteID {
```

Comment:
Does that mean that local users (`participant.GetRemoteID() == ""`) are always added?